src/services/loading-files.js

```diff
@@ -41,29 +41,43 @@ export function createExecutableFromProject(project) {
  */
 export function createProjectStructure(mainFile) {
   let modules = findRequires(mainFile)
+  let orderedModNames = modules.map((m) => m.name)
 
   for (let i = 0; i < modules.length; i++) {
-    if (modules[i].content || !fs.existsSync(modules[i].path)) continue
-
     modules[i].content = fs.readFileSync(modules[i].path, 'utf-8')
 
     const requiresInMod = findRequires(modules[i].content)
+
     requiresInMod.forEach((mod) => {
       const existingMod = modules.find((m) => m.name === mod.name)
-      if (existingMod) {
-        modules = modules.filter((m) => m.name !== existingMod.name)
+      if (!existingMod) {
+        modules = modules.push(mod)
```

Comment:
This sets the modules variable to the result of modules.push(), which is the new length of the array.
```suggestion
        modules.push(mod)
```